File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditUrlNameEventDto.java
package de.helfenkannjeder.helfomat.api.organization.event;

import de.helfenkannjeder.helfomat.core.organization.OrganizationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditUrlNameEventDto extends OrganizationEventDto {
    private String urlName;

    OrganizationEditUrlNameEventDto() {
    }

    public OrganizationEditUrlNameEventDto(OrganizationId organizationId, String urlName) {
        super(organizationId);
        this.urlName = urlName;
    }

    public String getUrlName() {
        return urlName;
    }

    @Override
    public <T> T visit(OrganizationEventDtoVisitor<T> visitor) {
        return visitor.visit(this);
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditUrlNameEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditUrlNameEventDto(
    override val organizationId: OrganizationId,
    val urlName: String,
    override val eventApplicable: EventApplicability
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}